### ISSUE TITLE
Feature Request: Add obfuscation option

### ISSUE THREAD

**Issue by GPXCoder** [sabrogden/Ditto#635]
I often work with multiple passwords and sometimes have to screenshare. Would it be possible to obfuscate all but the first 'x' characters on the displayed clip data?

Ideally, the number of un-obfuscated characters would be set in the options.

**Comment by lheydon** [comment]
**@GPXCoder**, it is a good idea for a feature to add generally, but I'd say that if you are saving your passwords somewhere completely unsecure like Ditto then you probably need to rethink that in the first place! 😲

**Comment by Chealer** [comment]
@GPXCoder: This is an ITS. Each ticket should report an issue, not just request specific solutions.

I assume you consider that what ticket #325 requests is not an optimal solution to your issue? Are you intentionally storing these passwords in Ditto?

[CopyQ 12 and ulterior has a preference to hide itself from screencasts](https://github.com/hluk/CopyQ/issues/3165), but I doubt it affects screen sharing.

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by GPXCoder** [comment]
@Chealer 
Thank you for responding. I use a password manager, I've configured Ditto not to add entries from the app, great feature btw. Occasionally I see passwords appear, I suspect they're appearing due to features such as Screen Connect's shared clipboard feature.

Ref #325 Whether a password manager feature is a scope creep to far, is not for me to say, but I feel that obfuscation is probably easier to implement and more inline with Dittos core feature set. Additionally, I believe that these solutions solve different problems.

**Comment by Chealer** [comment]
@GPXCoder:
> [@Chealer](https://github.com/Chealer) Thank you for responding. I use a password manager, I've configured Ditto not to add entries from the app, great feature btw. Occasionally I see passwords appear, I suspect they're appearing due to features such as Screen Connect's shared clipboard feature.

Please clarify what you mean by "Screen Connect". Did you try [excluding applications](https://github.com/sabrogden/Ditto/issues/947) to identify which one(s) is/are the source(s)?
 
> Ref [#325](https://github.com/sabrogden/Ditto/issues/325) Whether a password manager feature is a scope creep to far, is not for me to say, but I feel that obfuscation is probably easier to implement and more inline with Dittos core feature set. Additionally, I believe that these solutions solve different problems.

Sorry, I was not asking about optimality regarding use of development resources. I was asking if allowing certain entries to be encrypted would solve the issue you are experiencing at least as well as what you request here.

**Comment by GPXCoder** [comment]
Ref Screen connect, Note: This is not really relevant, but I'm happy to answer. While I screen share as part of my job, I also remotely manage servers and end-user devices using remote access tools. The one I currently use for work is called Screen Connect. If I'm working with a colleague and we are both remotely connected to the same device, all three clipboards are shared. i.e. I can copy and paste from the remote device to the local machine. Therefore, if my colleague uses the clipboard to paste a password, it will be added to my Ditto clipboard history. I would rather that Ditto retain the history of items copied from remote devices, as I may have many snippets I need to paste into documentation or reports.

No encryption is not the goal for me. My goal is to comfortably use Ditto functions while sharing my screen without worrying about a rogue password being displayed in clear text. If someone gains access to the ditto database on my device, I already have much bigger problems.

**Comment by Chealer** [comment]
@GPXCoder:
> Ref Screen connect, Note: This is not really relevant, but I'm happy to answer. While I screen share as part of my job, I also remotely manage servers and end-user devices using remote access tools. The one I currently use for work is called Screen Connect.

Do you mean ScreenConnect (previously ConnectWise ScreenConnect)?

> No encryption is not the goal for me. My goal is to comfortably use Ditto functions while sharing my screen without worrying about a rogue password being displayed in clear text. If someone gains access to the ditto database on my device, I already have much bigger problems.

I understand that encryption is not your _goal_, but would it be a good solution to your problem?

**Comment by GPXCoder** [comment]
I do not believe it would be a good solution to my problem as, so far as my understanding goes, encrypted items would still be visible in plain text while in use.

**Comment by Chealer** [comment]
What ticket #325 requests is that protected entries are "decrypted just at the moment of paste", so they would not "be visible in plain text", no. So, would that solve your issue well?

Please provide reproduction steps for the issue with "Screen Connect"(?)
